### PR TITLE
bakudan(C4): cleanup and documentation

### PR DESCRIPTION
**Pull request by teopiaz**
I did some renaming for the C4 actor and added some documentation the best I could.

Not 100% about how pMtx is used. But for sure is used for when the player attach the c4 to something that is moving like an enemy.

Can I safely remove the address of some variables? for example
```
int bakudan_count_8009F42C = 0;
int time_last_press_8009F430 = 0;
```

I guess that now is ok for the main exe but not yet ok for the overlays

### CODE REVIEW

File: src/Bullet/bakudan.h [hunk 1]
```diff
@@ -17,13 +17,13 @@ typedef struct BakudanWork
...
     MATRIX         field_C0_light_mtx[2];
-    MATRIX        *field_100_pMtx;
-    SVECTOR       *field_104;
-    int            field_108;
-    int            field_10C;
-    GV_PAD        *field_110_pPad;
-    int            field_114;
-    int            field_118;
+    MATRIX        *field_100_pMtx;         // (assigned if the c4 is placed on a moving target)
+    SVECTOR       *rotation;               // rotation vector (assigned if the c4 is placed on a moving target)
```
Comment (KieronJ) [inline src/Bullet/bakudan.h:21]: This is a position, the matrix is used for the rotation.

Comment (teopiaz) [reply src/Bullet/bakudan.h:21]: I got a bit confused (and partially still I am)
Reading more about the gte I saw that the MATRIX struct is defined as
```
typedef struct {
    short m[3][3];  /* 3x3 rotation matrix */
    long  t[3];     /* transfer vector */
} MATRIX;
```
but still this is equivalent to a 4x4 transformation matrix with the last row [0, 0, 0, 1]

```
m00 m01 m02 t1
m10 m11 m10 t2
m20 m21 m22 t3
0   0   0   1
```

so probably field_100_pMtx is not just the rotation matrix but the full transformation matrix

Comment (KieronJ) [reply src/Bullet/bakudan.h:21]: There's a seperate vector for position for some reason.

**Comment by KieronJ** [comment]
Thanks, looks good.

**Pull request merged**